com/paypal/Perfection.java

```diff
@@ -0,0 +1,42 @@
+package com.paypal;
+
+public class Perfection {
+
+    private static Perfection perf;
+
+    public synchronized static Perfection getPerf() {
+        if (perf == null) {
+            perf = new Perfection();
+        }
+        return perf;
+    }
+
+
+    public static boolean isPerfect(long candidate) {
+        boolean retVal;
+        long[] divisors = GetDivisors(candidate);
+        int sum = 0;
+        for (int d = 0; d < 1000; d++)
+        {
+            sum = sum + divisors[d];
+        }
+        if (sum == candidate)
+            retVal = true;
+        return retVal;
+    }
+
+
+    private static long[] GetDivisors(long candidate) {
+        long[] divisors = new long[];
+        int d = 0;
+        for (long i = 0; i < candidate; i++) {
```

Comment:
you don't need to for loop every number between 0 and candidate because the bigger possible divisor of a pair canditate is canditate/2 and for a even canditate is a (candidate-1)/2